### PR TITLE
Miscellaneous e2e tests improvements

### PR DESCRIPTION
**Pull request by Molter73**
**What type of PR is this?**

/kind cleanup

/kind documentation

/kind feature

**Any specific area of the project related to this PR?**

/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:
This PR makes a few improvements to the e2e tests:
- Fix a threading issue that prevented the tests from properly finishing when the last test threw an exception.
- Modify the `run-tests.sh` script so that parameters can be passed to `pytest`
- Add flags to skip tests using specific drivers.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/README.md [hunk 1]
```diff
@@ -46,12 +46,55 @@ entire repo as root, you can use the following commands instead:
...
+```
+
+Another common option could be to stop at the first failure:
+
+```sh
+mkdir -p build && cd build
+cmake -DCREATE_TEST_TARGETS=ON -DBUILD_LIBSCAP_MODERN_BPF=ON -DUSE_BUNDLED_DEPS=ON -DBUILD_BPF=ON -DBUILD_DRIVER=ON ..
+make sinsp-example driver bpf
+mkdir -p report/
+sudo ../test/e2e/scripts/run_tests.sh -x ../test/e2e/tests/
```
Comment (FedeDP) [inline test/e2e/README.md:75]: :heart:

**Review by FedeDP — APPROVED** [review]
/approve

**Review by LucaGuerra — APPROVED** [review]
Thanks as always @Molter73 , LGTM!

**Pull request merged**